### ISSUE TITLE
Magic number is reported for default parameter value

### ISSUE THREAD

**Issue by magro** [scalastyle/scalastyle#99]
I have a constructor with a default parameter value like this

``` java
class Foo(val x: Int = 42) {...}
```

When running scalastyle with MagicNumberChecker enabled, it reports the default value as magic number.

It would be great if this could be enhanced, e.g. that default parameter values are not reported at all or that only magic numbers are reported that occur more often than a given number of times.

**Comment by satendrakumar06** [comment]
Same for method. 

`def foo(x: Int = 42) {...}`

**Comment by satendrakumar06** [comment]
I have method for generating substring like:

```
  def getSubstring(line: String, beginIndex:Int, endIndex: Int): String =
    line.substring(beginIndex, endIndex)
```

**case 1**
`getSubstring(line, 2, 45)`

**case 2**
`getSubstring(line, beginIndex = 2, endIndex = 45)`

case 1 is warning. I agree. But case 2 should not be warning.(Correct me if i am wrong) 

@matthewfarwell Could please help me out why case 2 is waning ?

**Comment by Lasering** [comment]
This should be the default behavior, since using named parameters gives a description to the magic number.

**Comment by legopiraat** [comment]
Agreed with @satendrakumar06 case2 should not be a warning still want this implemented.

**Comment by matthewfarwell** [comment]
A PR would be great...

**Comment by jon-morra-zefr** [comment]
Would love this as well.  We use a lot of default parameters and without this I can't use this check.

**Comment by matthewfarwell** [comment]
Again, a PR would be great...
